utils/location.py

```diff
@@ -113,29 +113,39 @@
 cache = {}
 
 
+def geocode_location_string(location_string):
+    """
+    Requests Nomatim to geocode an input string. All results are cached and
+    reused.
+    """
+    location = None
+    if location_string in cache:
+        location = cache[location_string]
+    else:
+        logging.info(f"Calling geocoder with {location_string}...")
+        location = geolocator.geocode(location_string, addressdetails=True)
+        cache[location_string] = location
+    return location

```

Comment:
I would suggest using the more Pythonic `@lru_cache(maxsize=None)` function decorator to cache values returned by this function https://docs.python.org/3/library/functools.html#functools.lru_cache